src/scikit_build_core/builder/get_requires.py

```diff
@@ -66,6 +67,21 @@ def _load_scikit_build_settings(
     return SettingsReader.from_file("pyproject.toml", config_settings).settings
 
 
+class RootPath(type(Path())):  # type: ignore[misc]
```

Comment:
Hmm older pythons don't seem to like
```python
class RootPath(Path)
```
It's a bit ugly but it technically does the job.

Reply:
It's fine to make this a wrapper (`class RootPath:` with a data member, or even a `str` subclass). `Path` is weird since it can be different things based on what OS you are on.